// pdf-page-counter/lib/pdf-page-counter.js
let PDFJS = null

const DEFAULT_OPTIONS = {
    max: 0,
    //check https://mozilla.github.io/pdf.js/getting_started/
    version: 'v2.4.456'
}

async function PDF(dataBuffer, options) {
    let ret = {
        numpages: 0,
        info: null,
        metadata: null,
        text: "",
        version: null
    };

    if (typeof options == 'undefined') options = DEFAULT_OPTIONS;
    if (typeof options.max != 'number') options.max = DEFAULT_OPTIONS.max;
    if (typeof options.version != 'string') options.version = DEFAULT_OPTIONS.version;
    if (options.version == 'default') options.version = DEFAULT_OPTIONS.version;

    PDFJS = PDFJS ? PDFJS : require(`./pdf.js/${options.version}/build/pdf.js`);

    ret.version = PDFJS.version;

    // Disable workers to avoid yet another cross-origin issue (workers need
    // the URL of the script to be loaded, and dynamically loading a cross-origin
    // script does not work).
    PDFJS.disableWorker = true;
    let doc = await PDFJS.getDocument(dataBuffer);
    ret.numpages = doc.numPages;

    doc.destroy();

    return ret;
}

module.exports = PDF;
